src/clientlist.cpp

```diff
@@ -399,6 +399,16 @@ void unmanage_client(Window win) {
     if (m) monitor_apply_layout(m);
     ewmh_remove_client(win);
     tag_set_flags_dirty();
+
+    // Get the current client and update the windows focus.
+    client = frame_focused_client(tag->frame);
+    if (client) {
+        client_window_focus(client);
```

Comment:
I'm a bit confused about what this is supposed to do - aren't you essentially focusing the already focused client here?